Codecademy/docs
# Switched from CircleCI to Github Actions

## Overview

Switches from an 87-line CircleCI config to a 16-line Github Actions (GHA) config. 🔪 

Context: I'm getting us onto standard GHA actions using their new [composite actions](https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/) ([docs](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)) such as our own new [`Codecademy/run-on-yarn`](https://github.com/Codecademy/run-on-yarn). Much simpler configs and better integrated with Github itself.

### Notes

The Docker image used in CircleCI is a Cypress one, but we don't actually use Cypress / end-to-end testing in this repo. So `ubuntu-latest` or really anything with Node+Yarn is fine.

### Testing Instructions

1. Verify that the ❌ -d action runs for commits b42e401 and d7a8acb fail CI with errors related to their tasks: e.g. https://github.com/Codecademy/docs/runs/3965553939?check_suite_focus=true failing on `Type 'string' is not assignable to type 'number'.`
2. Verify that ff8caf0 passes all GHA runs ✅ : e.g. https://github.com/Codecademy/docs/runs/3966803955?check_suite_focus=true

Note that CircleCI is failing because it thinks we still want to run the same checks as on the `main` branch. 🤷
